### PR TITLE
Feat/add redirect

### PR DESCRIPTION
**Pull request by ehdi**

### CODE REVIEW

File: src/main/java/io/shelang/aghab/resource/RedirectResource.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    LinksService linksService;
+
+    @Blocking
+    @Route(path = "/r/:hash", methods = HttpMethod.GET)
+    public void redirect(RoutingContext rc) {
+        LinksDTO byHash = linksService.getByHash(
+                rc.request().getParam("hash"));
+        rc.response()
+                .putHeader("location", byHash.getUrl())
+                .setStatusCode(301)
```
Comment (alimalek71) [inline src/main/java/io/shelang/aghab/resource/RedirectResource.java:24]: By using 301 we redirect links **Permanently** and if a person clicks more than 1 time on the link the browser or other internet proxies may cache it. Do we what this happen? this has an effect on analytics, link redirection expiring, and disabling link.

Comment (alimalek71) [reply src/main/java/io/shelang/aghab/resource/RedirectResource.java:24]: I think we could save redirection code in the database for each link too, what do you think for future updates

Comment (ehdi) [reply src/main/java/io/shelang/aghab/resource/RedirectResource.java:24]: > By using 301 we redirect links **Permanently** and if a person clicks more than 1 time on the link the browser or other internet proxies may cache it. Do we what this happen? this has an effect on analytics, link redirection expiring, and disabling links.  
> 

I think it's very good for high throughput, in my humble opinion we can set it optional and use 301 for normal links and 302 for the links that need an expiration, etc.

Comment (ehdi) [reply src/main/java/io/shelang/aghab/resource/RedirectResource.java:24]: > I think we could save redirection code in the database for each link too, what do you think for future updates

Could you please explain it more?

Comment (alimalek71) [reply src/main/java/io/shelang/aghab/resource/RedirectResource.java:24]: > I think it's very good for high throughput, in my humble opinion we can set it optional and use 301 for normal links and 302 for the links that need an expiration, etc.

I agree with you, using both of them could help us.

> Could you please explain it more?

Adding a new column that indicates redirecting status codes.

**Pull request merged**